dumpling/metrics.go

```diff
@@ -30,11 +33,19 @@ var (
 		}, []string{"task", "source_id"})
 )
 
-// RegisterMetrics registers metrics.
+// RegisterMetrics registers metrics and saves the given registry for later use.
 func RegisterMetrics(registry *prometheus.Registry) {
 	registry.MustRegister(dumplingExitWithErrorCounter)
+	export.InitMetricsVector(prometheus.Labels{"task": "", "source_id": ""})
+	export.RegisterMetrics(registry)
 }
 
-func (m *Dumpling) removeLabelValuesWithTaskInMetrics(task string) {
-	dumplingExitWithErrorCounter.DeleteAllAboutLabels(prometheus.Labels{"task": task})
+func (m *Dumpling) removeLabelValuesWithTaskInMetrics(task, source string) {
+	labels := prometheus.Labels{"task": task, "source_id": source}
+	dumplingExitWithErrorCounter.DeleteAllAboutLabels(labels)
+	failpoint.Inject("SkipRemovingDumplingMetrics", func(_ failpoint.Value) {
```

Comment:
Why do we need this here?

Reply:
added a failpoint, otherwise integration test will remove all metrics so we can't check